crates/astria-bridge-withdrawer/src/bridge_withdrawer/startup.rs

```diff
@@ -0,0 +1,564 @@
+async fn get_allowed_fee_asset_ids(
+    client: sequencer_client::HttpClient,
+    state: Arc<State>,
+) -> eyre::Result<AllowedFeeAssetsResponse> {
+    let res = tryhard::retry_fn(|| client.get_allowed_fee_assets())
+        .with_config(make_sequencer_retry_config(
+            state.clone(),
+            "attempt to get allowed fee assets from Sequencer; retrying after backoff".to_string(),
+        ))
+        .await
+        .wrap_err("failed to get allowed fee asset ids from Sequencer after a lot of attempts");
+
+    state.set_sequencer_connected(res.is_ok());
+
+    res
+}
+
+#[instrument(skip_all)]
+async fn get_latest_balance(
+    client: sequencer_client::HttpClient,
+    state: Arc<State>,
+    address: Address,
+) -> eyre::Result<BalanceResponse> {
+    let res = tryhard::retry_fn(|| {
+        // only retry on tendermint_rpc errors, not deserialization or native conversion
+        let client = client.clone();
+        async move {
+            match client.get_latest_balance(address).await {
+                Ok(res) => Ok(Ok(res)),
+                Err(err) => {
+                    if let Some(tendermint_err) = err.as_tendermint_rpc() {
+                        Err(tendermint_err.inner().clone())
+                    } else {
+                        Ok(Err(err))
+                    }
+                }
+            }
+        }
+    })
+    .with_config(make_cometbft_retry_config(
+        state.clone(),
+        "attempt to get latest balance from CometBFT; retrying after backoff".to_string(),
+    ))
+    .await
+    .wrap_err("failed to get latest balance from Sequencer after a lot of attempts");
+
+    let res = res?.wrap_err("failed to deserialize the latest balance response from CometBFT");
+
+    // set cometbft as connected if received a response
+    state.set_sequencer_connected(res.is_ok());
+
+    res
+}
+
+fn make_cometbft_retry_config(
+    state: Arc<State>,
+    retry_message: String,
+) -> tryhard::RetryFutureConfig<
+    ExponentialBackoff,
+    impl Fn(u32, Option<Duration>, &tendermint_rpc::Error) -> futures::future::Ready<()>,
+> {
+    tryhard::RetryFutureConfig::new(u32::MAX)
+        .exponential_backoff(Duration::from_millis(100))
+        .max_delay(Duration::from_secs(20))
+        .on_retry(
+            move |attempt: u32, next_delay: Option<Duration>, error: &tendermint_rpc::Error| {
+                let state = Arc::clone(&state);
+                state.set_cometbft_connected(false);
+
+                let wait_duration = next_delay
+                    .map(humantime::format_duration)
+                    .map(tracing::field::display);
+                warn!(
+                    attempt,
+                    wait_duration,
+                    error = error as &dyn std::error::Error,
+                    retry_message,
+                );
+                futures::future::ready(())
+            },
+        )
+}
+
+fn make_sequencer_retry_config(
+    state: Arc<State>,
```

Comment:
Honestly think this is unnecessary when you run the startup logic. Either this returns successfully or it doesn't. If it returns successfully you know that you are connected.